File: backend/src/main/java/com/example/mytimeline/model/Comment.java
package com.example.mytimeline.model;

import java.time.LocalDateTime;

/**
 * comments テーブルの 1 行に対応するモデル。
 *
 * <p>{@link Post} と同じく、MyBatis が結果セットから組み立てるため引数なしコンストラクタと
 * setter を持つ可変クラスにしている（record は使えない）。</p>
 *
 * <p>{@link #author} は comments 単体には存在しない列で、users を JOIN したときだけ埋まる。
 * コメント一覧は必ず投稿者名を表示するので、コメントを取得する SQL では常に JOIN している。</p>
 */
public class Comment {

    private Long id;
    private Long postId;
    private Long userId;
    private String body;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private User author;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getPostId() {
        return postId;
    }

    public void setPostId(Long postId) {
        this.postId = postId;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User author) {
        this.author = author;
    }
}

File: frontend/src/types/comment.ts
/**
 * コメント API の型定義。バックエンドの DTO
 * (backend/src/main/java/com/example/mytimeline/dto) と 1:1 で対応させる。
 */

import type { PostAuthor } from './post'

export interface CommentResponse {
  id: number
  postId: number
  body: string
  /** 投稿と同じ形なので PostAuthor を使い回す */
  author: PostAuthor
  /** LocalDateTime のためタイムゾーンを持たない ISO 文字列 */
  createdAt: string
  updatedAt: string
}

/** コメントの作成・編集で送る内容。どちらも本文だけなので共用する */
export interface CommentRequest {
  body: string
}

export interface CommentListResponse {
  comments: CommentResponse[]
  /**
   * 次ページのカーソル。以降が無ければ null。
   *
   * タイムラインと違いコメントは古い順なので、続きは「このカーソルより新しいもの」になる。
   */
  nextCursor: number | null
}
